pkg/endpoint/events.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endpoint
+
+import (
+	"sync"
+
+	"github.com/cilium/cilium/pkg/eventqueue"
+)
+
+// EndpointRegenerationEvent contains all fields necessary to regenerate an endpoint.
+type EndpointRegenerationEvent struct {
+	owner        Owner
+	regenContext *regenerationContext
+	ep           *Endpoint
+}
+
+// Handle handles the regeneration event for the endpoint.
+func (ev *EndpointRegenerationEvent) Handle(res chan interface{}) {
+	e := ev.ep
+	owner := ev.owner
+	regenContext := ev.regenContext
+
+	err := e.RLockAlive()
```

Comment:
This acquires the endpoint lock, which I am not a fan of as I'd like to keep the critical section for regeneration to a minimum amount of time. I think that with the eventqueue we may be able to remove this in the future since no events can be enqueued for an endpoint if its EventQueue has been stopped.